recipes/tar/all/conanfile.py

```diff
@@ -1,98 +1,109 @@
-from conans import AutoToolsBuildEnvironment, ConanFile, tools
-from conans.errors import ConanInvalidConfiguration
 import os
 
-required_conan_version = ">=1.33.0"
+from conan import ConanFile, conan_version
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.env import VirtualBuildEnv
+from conan.tools.files import apply_conandata_patches, copy, get, replace_in_file, rmdir
+from conan.tools.gnu import Autotools, AutotoolsToolchain
+from conan.tools.layout import basic_layout
+from conan.tools.microsoft import is_msvc
+
+required_conan_version = ">=1.53.0"
 
 
 class TarConan(ConanFile):
     name = "tar"
     description = "GNU Tar provides the ability to create tar archives, as well as various other kinds of manipulation."
-    topics = ("tar", "archive")
     license = "GPL-3-or-later"
-    homepage = "https://www.gnu.org/software/tar/"
     url = "https://github.com/conan-io/conan-center-index"
-    settings = "os", "arch", "compiler", "build_type"
-
-    _autotools = None
+    homepage = "https://www.gnu.org/software/tar/"
+    topics = "archive"
 
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
+    package_type = "application"
+    settings = "os", "arch", "compiler", "build_type"
 
     @property
     def _settings_build(self):
         return getattr(self, "settings_build", self.settings)
 
     def configure(self):
-        del self.settings.compiler.libcxx
-        del self.settings.compiler.cppstd
+        self.settings.rm_safe("compiler.libcxx")
+        self.settings.rm_safe("compiler.cppstd")
+
+    def layout(self):
+        basic_layout(self, src_folder="src")
 
-    def requirements(self):
-        self.requires("bzip2/1.0.8")
-        self.requires("lzip/1.21")
-        self.requires("xz_utils/5.2.5")
+    def package_id(self):
+        del self.info.settings.compiler
 
     def validate(self):
         if self.settings.os == "Windows":
             raise ConanInvalidConfiguration("This recipe does not support Windows builds of tar")  # FIXME: fails on MSVC and mingw-w64
-        if not self.options["bzip2"].build_executable:
-            raise ConanInvalidConfiguration("bzip2:build_executable must be enabled")
-
-    def package_id(self):
-        del self.info.settings.compiler
+        if conan_version.major >= 2:
+            # Check does not work with Conan v2
+            if not self.dependencies.build["bzip2"].options.build_executable:
+                raise ConanInvalidConfiguration("bzip2:build_executable must be enabled")
+
+    def build_requirements(self):
```

Comment:
I don't see it as build requirement, but regular requires instead. In case you build with `-o */*:shared=True`, it will fail in runtime, because there is no other libraries available.